### DISCUSSION TITLE
Micro Swiss Drive

### DISCUSSION THREAD

**Discussion by chaser724** [Jyers/Marlin#344] [Q&A] [answered]
I was curious if anyone had made an edit to this for the micro swiss drive. I tried my hands at it but failed each time. Offsets are different and nozzle runs off bed for first pass. If not ill just stick to the smith 3d firmware. Just defiantly preferred this one.

**Comment by vhristenco** [comment, marked as answer]
For the Micro Swiss Direct Drive you have to set x_min_pos -5 and y_min_pos -10, as per their installation guide.

**Comment by Jyers** [comment]
Did manage to get this working?

**Reply by OldSaltyPossum** [reply]
I was successful editing the UBL-BLTouch-10x10 to work with the Micro-Swiss Direct Drive, All Metal Hotend and Petsfang Blokhead Duct. I taped my build plate, identified the center, ran the per-compiled 1.3.0 firmware, moved the nozzle to the center point, noted offsets (mine are x -10 and y -13.5). I then compiled the firmware using these figures, installed and selected manual bed leveling then the center point and verified the nozzle actually on the center of my plate. Next I marked the location of my BLTouch, measured offsets from my center point and added those offsets (mine are x -48 y -6). Finally selected Auto Home and verified the BLTouch contacted the center of the plate. Thanks Jyers for the source code and templates!

**Reply by AnyMouse835** [reply]
So am I understanding this correctly - the only way yo sort out the X,Y home issue with a Micro Swiss Direct Drive, is to build a version of the firmware with those offsets?

Is it not possible to implement setting them in the UI?

**Comment by smooreace** [comment]
Microswiss tells you what needs done through compiling because that where
it has to be done.

**Reply by AnyMouse835** [reply]
Ok, thanks. Unfortunately, I don't have the kit to compile a build on

**Reply by Cal-LF** [reply]
Jyres offers a compile service for a monthly fee. Give that a try.

**Comment by smooreace** [comment]
Not understanding exactly what you mean there, but it stupid simple. Its
nothing more than editing a text file and clicking a checkmark.

**Reply by Cal-LF** [reply]
Compiling FW is intimidating when first learning. Turns out to be easy once the VS Code compiler is installed and set up with the proper plugins.

**Reply by AnyMouse835** [reply]
Wouldn't know where to even start...

**Reply by AnyMouse835** [reply]
> Compiling FW is intimidating when first learning. Turns out to be easy once the VS Code compiler is installed and set up with the proper plugins.

Taking the v1.3.3 source, I can get that to build. 

The pre-built I've been using is E3V2-BLTouch-5x5-v4.2.2-v1.3.3

I can do the offsets in the base code - but it looks like there are a raft of other changes needed for BLTouch support and 5x5 probing

**Comment by smooreace** [comment]
There are at least a few hundred videos out there.

**Comment by scaramonga** [comment]
-8 -12 it was for me, using Satana unit, but have since reverted back, as DD is just way too heavy for the unit to perform as it was. Even adding dual Z, it does not do nothing for prints ;)

**Reply by AnyMouse835** [reply]
Good info - thanks
